drivers/sensor/icm20948/icm20948.c

```diff
@@ -0,0 +1,581 @@
+/*
+ * Copyright (c) 2023, Lazaro O'Farrill
+ * email: lazaroofarrill@gmail.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+
+#define DT_DRV_COMPAT invensense_icm20948
+
+#include "zephyr/kernel.h"
+#include "zephyr/sys/util.h"
+#include "stdint.h"
+#include "zephyr/drivers/i2c.h"
+#include "zephyr/drivers/sensor.h"
+#include "zephyr/init.h"
+#include "zephyr/logging/log.h"
+#include "zephyr/sys/__assert.h"
+#include "icm20948.h"
+
+/* Created by Lazaro O'Farrill on 09/03/2023. */
+
+LOG_MODULE_REGISTER(ICM20948, CONFIG_SENSOR_LOG_LEVEL);
+
+static int icm20948_bank_select(const struct device *dev, uint8_t bank)
+{
+	if (bank > 3) {
+		return -EINVAL;
+	}
+	const struct icm20948_config *cfg = dev->config;
+	const uint8_t reg_bank_mask = 0x30;
+
+	int ret =
+		i2c_reg_update_byte_dt(&cfg->i2c, ICM20948_REG_BANK_SEL, reg_bank_mask, bank << 4);
+
+	if (ret) {
+		LOG_ERR("Error switching banks");
+		return ret;
+	}
+
+	return 0;
+}
+
+static void icm20948_convert_temp(struct sensor_value *val, int16_t raw_val)
+{
+
+	/* Offset by 21 degrees Celsius */
+	int64_t in100 =
+		(raw_val * 100) + (ICM20948_ROOM_TEMP_OFFSET_DEG * ICM20948_TEMP_SENSITIVITY_X100);
+
+	/* Whole celsius */
+	val->val1 = in100 / ICM20948_TEMP_SENSITIVITY_X100;
+
+	/* Micro celsius */
+	val->val2 = ((in100 - (val->val1) * ICM20948_TEMP_SENSITIVITY_X100) * INT64_C(1000000)) /
```

Comment:
`(val->val1)` is the parenthesis misplaced?

Reply:
Yes it was. It wasn't needed anymore when refactoring the wrapper functions.